plaso/parsers/windefenderhistory.py

```diff
@@ -0,0 +1,429 @@
+
+  _FILE_SIGNATURE = "Magic.Version"
+
+  _VALUE_DESCRIPTIONS = [
+      {0: "Threat identifier", 1: "Identifier"},
+      {0: "UnknownMagic1", 1: "Threat name", 4: "Category"},
+      {
+          0: "UnknownMagic2",
+          1: "Resource type",
+          2: "Resource location",
+          4: "Threat tracking data size",
+          5: "Threat tracking data",
+          6: "Last threat status change time",
+          12: "Domain user1",
+          14: "Process name",
+          18: "Initial detection time",
+          20: "Remediation time",
+          24: "Domain user2",
+      },
+  ]
+
+  _CATEGORY_NAME = {
+      0: "INVALID",
+      1: "ADWARE",
+      2: "SPYWARE",
+      3: "PASSWORDSTEALER",
+      4: "TROJANDOWNLOADER",
+      5: "WORM",
+      6: "BACKDOOR",
+      7: "REMOTEACCESSTROJAN",
+      8: "TROJAN",
+      9: "EMAILFLOODER",
+      10: "KEYLOGGER",
+      11: "DIALER",
+      12: "MONITORINGSOFTWARE",
+      13: "BROWSERMODIFIER",
+      14: "COOKIE",
+      15: "BROWSERPLUGIN",
+      16: "AOLEXPLOIT",
+      17: "NUKER",
+      18: "SECURITYDISABLER",
+      19: "JOKEPROGRAM",
+      20: "HOSTILEACTIVEXCONTROL",
+      21: "SOFTWAREBUNDLER",
+      22: "STEALTHNOTIFIER",
+      23: "SETTINGSMODIFIER",
+      24: "TOOLBAR",
+      25: "REMOTECONTROLSOFTWARE",
+      26: "TROJANFTP",
+      27: "POTENTIALUNWANTEDSOFTWARE",
+      28: "ICQEXPLOIT",
+      29: "TROJANTELNET",
+      30: "EXPLOIT",
+      31: "FILESHARINGPROGRAM",
+      32: "MALWARE_CREATION_TOOL",
+      33: "REMOTE_CONTROL_SOFTWARE",
+      34: "TOOL",
+      36: "TROJAN_DENIALOFSERVICE",
+      37: "TROJAN_DROPPER",
+      38: "TROJAN_MASSMAILER",
+      39: "TROJAN_MONITORINGSOFTWARE",
+      40: "TROJAN_PROXYSERVER",
+      42: "VIRUS",
+      43: "KNOWN",
+      44: "UNKNOWN",
+      45: "SPP",
+      46: "BEHAVIOR",
+      47: "VULNERABILTIY",
+      48: "POLICY",
+      49: "EUS",
+      50: "RANSOM",
+      51: "ASR",
+  }
+
+  _DEFINITION_FILE = os.path.join(
+      os.path.dirname(__file__), "detection_history.yaml"
+  )
+
+  def _ReadFileHeader(self, file_object):
+    """Reads the beginning of the file to verify it's a Defender History log.
+
+      Args:
+          file_object (file): file-like object.
+
+      Raises:
+          ParseError: if the file header cannot be read.
+    """
+    data_type_map = self._GetDataTypeMap("detection_history_value")
+    header, header_size = self._ReadStructureFromFileObject(
+        file_object, 0, data_type_map
+    )
+
+    data_type_map = self._GetDataTypeMap("detection_history_value")
+    _, guid_size = self._ReadStructureFromFileObject(
+        file_object, header_size, data_type_map
+    )
+    signature, _ = self._ReadStructureFromFileObject(
+        file_object, header_size + guid_size, data_type_map
+    )
+
+    if (
+        header.data_size != 8 or header.data_type != 8
+    ) and self._FILE_SIGNATURE not in signature.value_string:
+      raise errors.ParseError("Invalid header")
```

Comment:
maybe `The UTF-16 string "Magic.Version:1.2" at offset 0x30 might be a more reliable "signature".`

and define GetFormatSpecification, this also removes the need to change utmp.py

Reply:
Done